More unit tests

carlosms opened the pull request:
Part of #32.

Also contains a couple of fixes for bugs uncovered by the new tests.

smacker commented on server/handler/schema_test.go:
```diff
@@ -0,0 +1,65 @@
+		sqlmock.NewRows([]string{"table"}).
+			AddRow("foo"))
+
+	suite.mock.ExpectQuery("DESCRIBE TABLE.*").WillReturnRows(
```
I would expect `DESCRIBE TABLE foo` here to be sure the second query is correct.

smacker commented on server/handler/query.go:
```diff
@@ -183,7 +184,10 @@ func addLimit(query string, limit int) string {
 	}
 
-	query = strings.TrimRight(strings.TrimSpace(query), ";")
+	re := regexp.MustCompile(`\/\*(?s:.)*?\*\/`)
```
maybe better define & compile the regexp only once outside of the function?

carlosms commented:
Thanks @smacker. Since the feedback is not critical, I'll merge this PR as it is to fix the integration tests on Travis on master. I'll create a new PR shortly addressing the feedback.
